### DISCUSSION TITLE
v3 is it expected following the simple example in the docs with fetch causes TS error?

### DISCUSSION THREAD

**Discussion by michaelaflores** [TanStack/query#4149] [Q&A]
When following the v3 "basic" [example](https://github.com/TanStack/query/tree/v3/examples/basic), replacing the use of Axios with `fetch` results in a TS issue where  `error` is of type `unknown` (I assume actually using Axios works here). 

```tsx
import { useQuery } from 'react-query';

const getPopularDataById = async (id: string) => {
  const requestUrl =
    process.env.NEXT_PUBLIC_API_URL + `/session/company/popular/${id}`;
  const response = await fetch(requestUrl);
  if (!response.ok) {
    throw new Error('Network response was not ok');
  }
  return response.json();
};

function usePopularData(companyId: string) {
  return useQuery(
    ["mostPopular", companyId],
    () => getPopularDataById(companyId),
    {
      enabled: !!companyId,
    },
  );
}

function DashboardController() {
  const companyId = '1';
  const { isLoading, isError, data, error } = usePopularData(companyId);
  if (isLoading) return <p>Loading...</p>;

// TS error: Object is of type 'unknown'
  if (isError) return <p>Error: {error.message}</p>;
  return (
    <div className="mt-20 text-black">
      <h1>Dashboard test</h1>
      <p>test</p>
    </div>
  );
}

export default DashboardController;
```

I'm curious if this is expected when using `fetch` and following the examples in the [docs](https://react-query-v3.tanstack.com/guides/query-functions)? 

This is resolved by providing a type in the second argument to the `useQuery` generic but I was wondering why the docs by default don't mention anything other than that a _custom_ hook must specify the error type, when this is also true when just using `useQuery` directly as the docs show.

i..e this example has the same error 

```tsx
function DashboardController() {
  const companyId = '1';
  const { isLoading, isError, data, error } = usePopularData(companyId);
  if (isLoading) return <p>Loading...</p>;

// TS error: Object is of type 'unknown'
  if (isError) return <p>Error: {error.message}</p>;
  return (
function DashboardController() {
  const companyId = '1';
  const { isLoading, isError, data, error } = useQuery(
    ['mostPopular', companyId],
    () => getPopularDataById(companyId),
  );
  if (isLoading) return <p>Loading...</p>;

  if (isError) return <p>Error: {error.message}</p>;
  return (
    <div className="mt-20 text-black">
      <h1>Dashboard test</h1>
      <ChartView />
    </div>
  );
}
  );
}
```

**Comment by TkDodo** [comment]
The typescript docs have been updated for v4:
https://tanstack.com/query/v4/docs/typescript#typing-the-error-field

also, see: https://tkdodo.eu/blog/react-query-and-type-script

**Comment by Magnussmari** [comment]
## Morning Briefing - March 16, 2026

### 📊 Inbox Status
- **Total inbox items**: 5
- **New captures yesterday**: 0  
- **Stale alerts**: 5 items (all >7 days in inbox)

### 🔄 Stale Items Requiring Processing
All current issues are marked as stale:
- #1 GitOS initialized — first capture (18 days old)
- #3 The EU AI Act: Sovereignty Implications for Small Nations (16 days old)
- #4 Cognitive Debt in Automated Decision Systems (16 days old) 
- #5 Responsibility Fog in Algorithmic Decision-Making (16 days old)
- #6 Weekly Review - 2026-03-01 (15 days old)

### 💡 Synthesis Opportunity
**AI Governance Philosophy Cluster**: Issues #3, #4, and #5 from Feb 28 form a coherent theme around AI governance challenges, decision-making responsibility, and cognitive debt. These could be synthesized into a Wiki page on "Algorithmic Accountability Challenges" with cross-references to sovereignty implications.

### 🎯 Recommended Actions
1. Process the Feb 28 AI governance cluster first (highest synthesis value)
2. Review and update issue #6 (Weekly Review)
3. Archive or update the initial GitOS capture (#1)

---
*Auto-generated morning briefing*
